test/built-ins/Atomics/wait/datablock-bufferdata-throws.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-map-objects
+description: A DataBlock value for bufferData throws a TypeError
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  1.Let buffer be ? ValidateSharedIntegerTypedArray(typedArray, true).
+    ...
+      9.If IsSharedArrayBuffer(buffer) is false, throw a TypeError exception.
+        ...
+          4.If bufferData is a Data Block, return false.
+features: [ TODO ]
+---*/
+
+//TODO: note this test is valid in an environments when [[CanBlock]] is true or false (valid in both)
+
+var int32Array = new Int32Array(new ArrayBuffer(1024));
+
+assert.throws(TypeError, () => Atomics.wait(int32Array, 0, 0, 0));
```

Comment:
In cases like this, you might want to poison the next arguments so you can assert the TypeError is thrown in the right time.

In this case, this line could be translated to:

```js
var poisoned = {
  get valueOf() { throw new Test262Error("Should not evaluate this code"); }
};
assert.throws(TypeError, function() {
  Atomics.wait(int32Array, poisoned, poisoned, poisoned);
});
```